hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java

```diff
@@ -616,10 +629,100 @@ public Fees calculateFees(@NonNull final FeeContext feeContext) {
         return feeContext
                 .feeCalculator(subType)
                 .addBytesPerTransaction(bpt)
+                .addRamByteSeconds(associationsRbs)
                 .addRamByteSeconds(rbs * USAGE_PROPERTIES.legacyReceiptStorageSecs())
                 .calculate();
     }
 
+    private long getAutoAssociationsRbsFee(final FeeContext feeContext) {
+        final var body = feeContext.body();
+        final var op = body.cryptoTransferOrThrow();
+        final var tokenTransfers = op.tokenTransfers();
+
+        // If the token transfers collection is empty, then we have only hbars transfer and no token
+        // associations are needed, so we do not need to calculate the next part.
+        if (tokenTransfers.isEmpty()) {
+            return 0;
+        }
+
+        final var accountStore = feeContext.readableStore(ReadableAccountStore.class);
+        final var tokenRelStore = feeContext.readableStore(ReadableTokenRelationStore.class);
+
+        final Set<AccountID> recipientsSet = getTransferRecipients(tokenTransfers);
```

Comment:
If all we are doing is finding if the token relation doesn't exist, we should be able to re-use some functionality from `validateAndBuildAutoAssociation` and count number of new associations that will be formed with this list. 

And then we need to do is add the rbs value in fees. 
```
numAutoAssociations * THREE_MONTHS_IN_SECONDS * CREATE_SLOT_MULTIPLIER;
```

Reply:
Yes, we need to find the new associations that need to be created. I am not sure if we can reuse even parts of the implementation in `validateAndBuildAutoAssociation` because `calculateFees` is called before `handle` and in `handle` we already have the hollow accounts that are part of the transfers but in `calculateFees` they are not created yet. Because of that the accountId is still null at that point and `tokenRelStore.get(accountId, tokenId)` will throw an exception.